### PR TITLE
Check errors in buildspec cmd and fix conflicting -out flag

### LINKED ISSUE #193: mage buildspec does not update supported.go
If you run `mage buildspec` it will run without errors and provide the expected log output, but it doesn't actually do anything.

`>> Buildspec from internal/spec/*.yml to internal/pkg/agent/program/supported.go`

The problem is that errors are occurring, but they go unchecked in the code. The root cause is a recent change to workaround the fact that `github.com/elastic/elastic-agent/dev-tools/license` uses a [conflicting `-out` flag](https://github.com/elastic/elastic-agent/blob/dfa71779592fc4a00f8a51aa9669407bb272d031/dev-tools/licenses/license_generate.go#L50). Due to this change, the `output` variable is never set. So the later `WriteFile` uses an empty file name.

### PR DESCRIPTION
**Pull request by andrewkroh**
Add missing error checks and implement workaround for the conflicting `-out` flag
used in dev-tools/licenses.

Fixes #193

### CODE REVIEW

**Comment by andrewkroh** [comment]
> This pull request does not have a backport label. Could you fix it @andrewkroh?

No, I cannot. I don't have permissions to apple labels.

**Review by ph — APPROVED** [review]
LGTM

I've tested it locally.

@andrewkroh Can you run a `mage buildspec` and commit the changes, the generated file will be different because of the new path of this repository.

**Comment by ph** [comment]
```diff
 // or more contributor license agreements. Licensed under the Elastic License;
 // you may not use this file except in compliance with the Elastic License.
 
-// Code generated by elastic-agent/internals/dev-tools/cmd/buildspec/buildspec.go - DO NOT EDIT.
+// Code generated by elastic-agent/internals/dev-tools/buildspec/buildspec.go - DO NOT EDIT.
 
 package program
 
@@ -25,7 +25,7 @@ func init() {
        // internal/spec/metricbeat.yml
        // internal/spec/osquerybeat.yml
        // internal/spec/packetbeat.yml
-       unpacked := packer.MustUnpack("eJzEelt3oziX9v33M/r2m5nmUE43s9Z7YZzmZIeUcSyE7pBwAFvC7hgfYNb891kSBwMmqUpVd78XXollIW1t7cOzn83//JJm+eYtC+mvx8OG/Boe2H8eN2/nzdt/FYz+8t+/YGbk6GUf
+       unpacked := packer.MustUnpack("eJzUWlt3o7iWfp+f0a9zaS7ldDNrnQfjNDc7pIxjSegNCQewBXbH+AKz5r/PkrgYMElVqmv6rPPglVgW0pa097e//Yn/+SXJ8s1bFrBfj4cN/TU4pP953LydN2//VaTsl//+haRGjl/2
        SupportedMap = make(map[string]Spec)
```

**Comment by andrewkroh** [comment]
I'm seeing `panic: Cannot read spec from internal/spec/filebeat.yml` when running ` go test ./internal/pkg/agent/program/`. 😢 .

**Comment by ph** [comment]
I will take a look, I am currently working on another PR that would require this.

File: internal/spec/filebeat.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ args: [
   "-E", "setup.template.enabled=false",
   "-E", "management.enabled=true",
   "-E", "logging.level=debug",
-  "-E", "gc_percent=${FILEBEAT_GOGC:100}"
+  "-E", "gc_percent=${FILEBEAT_GOGC:100}",
```
Comment (ph) [inline internal/spec/filebeat.yml:8]: o_O, that should have raised a check issue elsewhere.

Comment (ph) [reply internal/spec/filebeat.yml:8]: @narph This missing ',' is strange to me and also the line disabling the `modules` they are not present in the original file https://github.com/elastic/beats/blob/main/x-pack/elastic-agent/spec/filebeat.yml#L6-L9 ?

**Pull request merged**